lib/cloud/services/fileService.js

```diff
@@ -0,0 +1,105 @@
+var base64 = require('base64-stream');
+var path = require("path");
+var q = require("q");
+var through = require('through2');
+var os = require('os');
+var fs = require('fs');
+var multer = require('multer');
+
+var imageDir = os.tmpdir() + '/wfm';
+
+/**
+ * Create temporary storage folder used by mutler to store files before
+ * uploading to permanent storage
+ */
+function createTemporaryStorageFolder() {
+  fs.mkdir(imageDir, '0775', function(err) {
+    if (err && err.code !== 'EEXIST') {
+      console.log(err);
+      throw new Error(err);
+    }
+  });
+}
+
+/**
+ * Utility function for saving file in temp folder
+ * @param fileMeta
+ * @param stream
+ */
+function writeStreamToFile(fileMeta, stream) {
+  var deferred = q.defer();
+  stream.on('end', function() {
+    deferred.resolve(fileMeta);
+  });
+  stream.on('error', function(error) {
+    deferred.reject(error);
+  });
+  var filename = imageDir + '/' + fileMeta.uid;
+  stream.pipe(fs.createWriteStream(filename));
+  return deferred.promise;
+}
+
+/**
+ * Parse Base64 URL into stream that can be saved as file.
```

Comment:
I'm not sure how the base64 is being uploaded from the client. I thought it would be in a text file..